packages/backend/src/helpers/auth.ts

```diff
@@ -71,3 +71,22 @@ export async function getOrCreateUser(email: string): Promise<User> {
 
   return user
 }
+
+export async function validateAdminTokenAndGetUser(
+  token: string,
+): Promise<User | null> {
+  // NOTE: we use a different key to prevent a vuln where an end user can send
+  // their auth cookie in the `x-plumber-admin-token` header value to gain
+  // admin access.
+  const { userEmail } = jwt.verify(token, appConfig.adminJwtSecretKey) as {
+    userEmail: string
+  }
+
```

Comment:
with this code, it depends on jwt.verify to throw an error to restrict unauthorized access. 
```
  "errors": [
    {
      "message": "Context creation failed: jwt malformed",
      "code": "INTERNAL_SERVER_ERROR"
    }
  ]
}
```
For standardization, we should rely on graphql-shield to throw the same unauthorized error like a normal logged out user. i.e. 
```
  "errors": [
    {
      "message": "Not Authorised!",
      "code": "INTERNAL_SERVER_ERROR"
    }
```
One way is that we add a try catch block to jwt.verify and set isAdminQuery to false if an error is thrown.

Reply:
i think for extensibility, we can encode in the JWT to include perms.
e.g. 
```
{
  "userEmail": "example@test.com",
  "permissions": "read"
}
```